source/common/stats/stat_merger.cc

```diff
@@ -1,17 +1,84 @@
 #include "common/stats/stat_merger.h"
 
+#include <algorithm>
+
 namespace Envoy {
 namespace Stats {
 
-StatMerger::StatMerger(Stats::Store& target_store) : temp_scope_(target_store.createScope("")) {}
+StatMerger::StatMerger(Store& target_store) : temp_scope_(target_store.createScope("")) {}
+
+StatMerger::DynamicSpans StatMerger::DynamicContext::encodeComponents(StatName stat_name) {
+  DynamicSpans dynamic_spans;
+  uint32_t index = 0;
+  auto record_dynamic = [&dynamic_spans, &index](absl::string_view str) {
+    DynamicSpan span;
+    span.first = index;
+    index += std::count(str.begin(), str.end(), '.');
+    span.second = index;
+    ++index;
+    dynamic_spans.push_back(span);
+  };
+  SymbolTableImpl::Encoding::decodeTokens(
+      stat_name.data(), stat_name.dataSize(), [&index](Symbol) { ++index; }, record_dynamic);
```

Comment:
Does the lambda only get called for the dynamic portions? From a quick read this is a little confusing. Can you add some more comments?